test/test_utils.py

```diff
@@ -753,3 +753,141 @@ def test_match(self):
+        self.assert_equal_dict(expected, result)
+
+    def test_local_only(self):
+        base = {}
+        remote = {}
+        local = {"B": "local"}
+
+        expected = local
+        result = three_way_merge(local=local, base=base, remote=remote)
+        self.assert_equal_dict(expected, result)
+
+    def test_remote_only(self):
+        base = {}
+        remote = {"A": "remote"}
+        local = {}
+
+        expected = remote
+        result = three_way_merge(local=local, base=base, remote=remote)
+        self.assert_equal_dict(expected, result)
+
+    def test_key_removed_no_conflict(self):
+        base = {"C": "base"}
+        remote = {}
+        local = {}
+
+        expected = {}  # Should be empty as "C" was removed from both
+        result = three_way_merge(local=local, base=base, remote=remote)
+        self.assert_equal_dict(expected, result)
+
+    def test_key_removed_from_remote(self):
+        base = {"C": "base"}
+        remote = {}
+        local = {"C": "base"}
+
+        expected = remote  # The default on_conflict kwarg should cause remote to win
+        result = three_way_merge(local=local, base=base, remote=remote)
+        self.assert_equal_dict(expected, result)
+
+        # Now with the on_conflict param set to "local"
+
+        expected = local
+        result = three_way_merge(local=local, base=base, remote=remote, on_conflict="local")
+        self.assert_equal_dict(expected, result)
+
+    def test_key_removed_from_local(self):
+        base = {"C": "base"}
+        remote = {"C": "base"}
+        local = {}
+
+        expected = remote  # The default on_conflict kwarg should cause remote to win
+        result = three_way_merge(local=local, base=base, remote=remote)
+        self.assert_equal_dict(expected, result)
+
+        # Now with the on_conflict param set to "local"
+
+        expected = local
+        result = three_way_merge(local=local, base=base, remote=remote, on_conflict="local")
+        self.assert_equal_dict(expected, result)
+
+    def test_merge_no_potential_conflict(self):
+        base = {"C": "base"}
+        remote = {"A": "remote", "C": "base"}
+        local = {"B": "local", "C": "base"}
+
+        expected = {"A": "remote", "B": "local", "C": "base"}
+        result = three_way_merge(local=local, base=base, remote=remote)
+        self.assert_equal_dict(expected, result)
+
+    def test_merge(self):
+        """
+        Shows that a change in only one place does not consitute a conflict.
+        :return:
+        """
+        base = {"C": "base"}
+        remote = {"A": "remote", "C": "remote"}
+        local = {"B": "local", "C": "base"}
+
+        expected = {"A": "remote", "B": "local", "C": "remote"}
+        result = three_way_merge(local=local, base=base, remote=remote)
+        self.assert_equal_dict(expected, result)
+
+        # Now with local as the on_conflict winner
+        expected = {"A": "remote", "B": "local", "C": "remote"}
```

Comment:
Expected value is wrong here. It should be: expected = {"A": "remote", "B": "local", "C": "base"}. Thus logic in your code is wrong.

Reply:
As local has not changed the value of the key C, we can expect that even with the on_conflict value set to "local" we should get out a dictionary that includes the changed value.
I could be wrong, but I was working with the presumption that a conflict only exists if there is some sort of change between the base and the remote AND a differing change between the base and the local.